### ISSUE TITLE
Using API to checkin\checkout files to TFVC repos

### ISSUE THREAD

**Issue by zeppaman** [Microsoft/vsts-dotnet-samples#42]
Hi,
I need to automate some activities by using vsts apis. Basically I need to get files, alter them, and checkin. 
First part of the task is covered by TfvcHttpClient that allow me to access to the whole TFVC repository. 

There is lot of stuff for git repos,  but I cannot move to git. 

There is some resources I missed or can you address me to a solution?

**Comment by zeppaman** [comment]
Hi 
this issue can be closed as I got how this can be implemented.

Changes are made creating changsets, where I can add all the changes to include in the checkin operation.

Here a short snippet about how to do (sorry it is in Vb.Net):

```vb          Dim client = conn.GetClient(Of TfvcHttpClient)()
            Dim localPath As String = "C:\\...\\test.txt" ' local path
            Dim diskFile = New FileInfo(localPath)

            ' define the item (the file to be uploaded)
            Dim file As TfvcItem = New TfvcItem()
            file.Path = "$/DbStorage/prova_save_" & DateTime.Now.ToString("yyyyMMddHHmmss") & ".txt" ' define a remote path name
            file.ContentMetadata = New FileContentMetadata()

            file.ContentMetadata.Encoding = 65001

            ' all following settings seems not relevant 
            'file.ContentMetadata.ContentType = MimeMapping.GetMimeMapping(localPath)
            'file.ChangeDate = DateTime.Now
            'file.Size = diskFile.Length
            'file.Encoding = 65001
            'file.IsPendingChange = True
            'file.ChangesetVersion = 0
            'file.ContentMetadata.IsBinary = False
            'file.ContentMetadata.EncodingWithBom = False
            'file.ContentMetadata.Extension = ".txt"

            Dim change As TfvcChange = New TfvcChange(file, VersionControlChangeType.Add)
            change.NewContent = New ItemContent()
            change.NewContent.Content = Convert.ToBase64String(System.IO.File.ReadAllBytes(localPath))
            change.NewContent.ContentType = ItemContentType.Base64Encoded

            change.Item.Path = file.Path

            'create the changeset
            changeset = New TfvcChangeset()

            changeset.Comment = "AUTOMATICALLY ADDED FROM API"

            ' add changes list to the changeset
            Dim changesList As List(Of TfvcChange) = New List(Of TfvcChange)
            changesList.Add(change)
            changeset.Changes = changesList


            client.CreateChangesetAsync(changeset, project.Name).SyncResult()
```

Here some reference that bring me to the right solution

- https://visualstudio.uservoice.com/forums/330519-team-services/suggestions/16881988-vsts-tfs-rest-api-check-in-changes
- https://stackoverflow.com/questions/41703475/updating-a-file-using-rest-api-visual-studio-team-services


Please tell me if there is a bettere solution.

**Comment by vtbassmatt** [comment]
Really glad you got unblocked! At a quick glance, your code looks right. It would be good for us to have some TFVC samples, so I'll take this one.

**Comment by vtbassmatt** [comment]
Addressed with #48
